### ISSUE TITLE
archinstall.lib.exceptions.DiskError

### ISSUE THREAD

**Issue by matteoturini** [archlinux/archinstall#588]
Could not mount and check for content on /dev/sda1 because: b"mount /tmp/<super long uuid?>: unknown filesystem type 'swap'.

No filesystems as I wiped the partition tables and rebooted twice.

**Comment by matteoturini** [comment]
Should add a warning when it errors out like this: 

wipe partition tables:
wipefs -a /dev/sd<x>
reboot

**Comment by Gotchfutchian** [comment]
same issue here, i have no FS on my sda drive and it keeps throwing this error. 

im just gonna grab a temp distro until this is figured out becasue this laptop refuses to cooperate with arch.

**Comment by Torxed** [comment]
Apologies for the late answers, just back from vacation.
Could one of you attach your `install.log` file from `/var/log/archinstall`, and perhaps the output of `fdisk -l /device`?

**Comment by Allamuth** [comment]
I get the same error. I tried multiple times, with and without preexisting FS, always getting the same error.
Attached are the needed logs.

**LSBLK**
```
NAME     MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINTS
sdb        8:16   0 223.6G  0 disk  
sdb1       8:17   0   512M  0 part  /mnt/boot
sdb2       8:18   0 223.1G  0 part  
luksloop 254:0    0 223.1G  0 crypt 
```

**install.log**
[install.log](https://github.com/archlinux/archinstall/files/6763283/install.log)

**Comment by Ammako** [comment]
I get this if I use UEFI... on BIOS I instead get `New partition never showed up after adding new partition on Filesystem(blockdevice=BlockDevice(/dev/sda), mode=2) (timeout 10 seconds).`

I've had success manually creating the partitions, and mounting at /mnt, and using that, but then of course it runs into #541, so that doesn't help. I have to use /mnt, because re-using existing partitions on disk doesn't work, but that's a separate issue altogether.

**Comment by Ammako** [comment]
I have gotten past this DiskError exception by removing the arch iso version of archinstall, and replacing with latest git. Unfortunately I then later get an error about systemctl command not existing, but you might have better luck with it.

**Comment by matteoturini** [comment]
I fixed it by doing wipefs, partition the drive, wipefs and reboot.

Em sex., 16 de jul. de 2021 04:34, Ammako ***@***.***>
escreveu:

**Comment by l4zy0n3** [comment]
> I get this if I use UEFI... on BIOS I instead get `New partition never showed up after adding new partition on Filesystem(blockdevice=BlockDevice(/dev/sda), mode=2) (timeout 10 seconds).`
> 
> I've had success manually creating the partitions, and mounting at /mnt, and using that, but then of course it runs into #541, so that doesn't help. I have to use /mnt, because re-using existing partitions on disk doesn't work, but that's a separate issue altogether.
> 
> If using BIOS, it goes through, but doesn't boot, of course.
> 
> A real shame, because everything worked absolutely perfectly on my VM beforehand.
fixed by #578

**Comment by Torxed** [comment]
I think this issue is solved now, if not by #578 then probably by #426.
I'll close this issue, but if the issue persist feel free to re-open or ping me to re-open.
